src/AzureAppConfigurationImpl.ts

```diff
@@ -168,11 +175,49 @@ export class AzureAppConfigurationImpl extends Map<string, any> implements Azure
      */
     async load() {
         await this.#loadSelectedAndWatchedKeyValues();
-
         // Mark all settings have loaded at startup.
         this.#isInitialLoadCompleted = true;
     }
 
+    /**
+     * Construct hierarchical data object from map.
+     */
+    constructConfigurationObject(options?: ConfigurationObjectConstructionOptions): Record<string, any> {
+        const separator = options?.separator ?? ".";

```

Comment:
Should we have some restrictions on what is allowed as a separator?

Reply:
Good question. Do we allow an arbitrary character as separator in import/export? Or a string with multiple characters?

Reply:
Just checked import/export in portal, the separator supports an allowlist: 
<img width="351" alt="image" src="https://github.com/Azure/AppConfiguration-JavaScriptProvider/assets/2351748/1b3fd465-8a2a-4dba-8145-6ca4778b42e8">

For the "local.json -> import and flatten in azconfig-> restore hierarchy js provider" use case, it makes sense to restrict the allowed values and make it consistent with portal. But is it possible that we add more allowed separators in the future? If so, I would rather not to add the restriction. Anyway users have to specify the separator by their own. And I don't see potential problem there.

Reply:
We already have these restrictions in [CLI](https://learn.microsoft.com/en-us/cli/azure/appconfig/kv?view=azure-cli-latest#az-appconfig-kv-export), [portal and ADO pipeline tasks](https://msazure.visualstudio.com/Azure%20AppConfig/_git/SyncClient.TypeScript?path=/libraries/azure-app-configuration-sync/src/internal/constants.ts&version=GBmain&line=18&lineEnd=19&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents). If we ever add a new separator to the list, we'd have to make this change everywhere. I think it makes sense to follow the existing pattern from other clients and restrict the list of allowed separators.

Reply:
Ok, then it makes total sense.
 I’ll add the restriction. I can change type of “separator” from string to allowed values to benefit TS developers, and also add the restriction in the same time.

Reply:
restriction added via f4953da